### PR TITLE
WS: some cleanup (Java code, JavaDoc, JUnit method signatures, ...)

### PR DESCRIPTION
**Pull request by cbasener**
Several cleanups

- corrected some JavaDocs
- removed not needed casts
- usage of the diamond operator
- usage of constructor chaining in KeyResult; new test added
- deleted Exceptions, that are not thrown, from method signatures
- idiomatic use of isEmpty()
- idiomatic use of array brackets
- deleted not needed imports
- usage of better fitting JUnit assertions
- fail() instead of assertFalse(true)
- ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -305,7 +305,7 @@ The ability to reconnect on  the initial connection failure is _NOT_ an Options
 
 #### Clusters & Reconnecting
 
-The Java client will automatically reconnect if it loses its connection the nats-server. If given a single server, the client will keep trying that one. If given a list of servers, the client will rotate between them. When the nats servers are in a cluster, they will tell the client about the other servers, so that in the simplest case a client could connect to one server, learn about the cluster and reconnect to another server if its initial one goes down.
+The Java client will automatically reconnect if it loses its connection to the nats-server. If given a single server, the client will keep trying that one. If given a list of servers, the client will rotate between them. When the nats servers are in a cluster, they will tell the client about the other servers, so that in the simplest case a client could connect to one server, learn about the cluster and reconnect to another server if its initial one goes down.
```
Comment (scottf) [inline README.md:308]: Ok

File: src/examples/java/io/nats/examples/ConnectTime.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 /**
  * This example will demonstrate how to build a listener that will track connect time.
- *
+ * <p>
```
Comment (scottf) [inline src/examples/java/io/nats/examples/ConnectTime.java:22]: Not really necessary, but ok.

File: src/examples/java/io/nats/examples/autobench/JsPubAsyncRoundsBenchmark.java [hunk 3]
```diff
@@ -70,7 +70,7 @@ public void execute(Options connectOptions) throws InterruptedException {
                     for (long l = 0; l < thisRound; l++) {
                         futures.add(js.publishAsync(subject, payload));
                     }
-                    while (futures.size() > 0) {
+                    while (!futures.isEmpty()) {
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/JsPubAsyncRoundsBenchmark.java:73]: Changing code just to change code is unnecessary. I get that this is the preferred way, my ide tells me so all the time, it's probably more readable / expressive, but it's actually more expensive for most of the collections we use since `isEmpty` asks `size() == 0`

File: src/examples/java/io/nats/examples/autobench/JsPubBenchmark.java [hunk 4]
```diff
@@ -26,7 +26,7 @@ public class JsPubBenchmark extends AutoBenchmark {
     private static final Map<String, String> SAVED_SUBJECTS = new HashMap<>();
 
     public static String getKey(long messageCount, long messageSize) {
-        return "" + messageCount + "x" + messageSize;
+        return messageCount + "x" + messageSize;
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/JsPubBenchmark.java:29]: Non production code who cares? Yeah it's not optimal and if you are working in this file for some actual code change or fix, great, change this, but otherwise no.

File: src/examples/java/io/nats/examples/autobench/LatencyBenchmark.java [hunk 5]
```diff
@@ -43,7 +43,7 @@ public LatencyBenchmark(String name, long messageCount, long messageSize, String
         this.lcsv = lcsv;
     }
 
-    public void execute(Options connectOptions) throws InterruptedException {
+    public void execute(Options connectOptions) {
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/LatencyBenchmark.java:46]: Non client production code, only if already changing or fixing the class.

File: src/examples/java/io/nats/examples/autobench/LatencyBenchmark.java [hunk 6]
```diff
@@ -240,11 +240,11 @@ public double calcMedian() {
         measurements.sort(Long::compareTo);
 
         if (size % 2 == 1) {
-            return measurements.get(middle).longValue();
+            return measurements.get(middle);
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/LatencyBenchmark.java:243]: Non client production code, only if already changing or fixing the class.

File: src/examples/java/io/nats/examples/autobench/NatsAutoBench.java [hunk 7]
```diff
@@ -27,7 +27,7 @@
  * Implements a benchmark more like the .net client that loops over a number
  * of scenarios benchmarking each one. This class hard codes most settings to minimize
  * boilerplate and focus on the nats client code (plus measurement).
- * 
+ * <p>
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/NatsAutoBench.java:30]: Is this really important to change?

File: src/examples/java/io/nats/examples/autobench/NatsAutoBench.java [hunk 8]
```diff
@@ -62,7 +62,7 @@ public static void main(String[] args) {
                                                     connectionTimeout(Duration.ofSeconds(1)).
                                                     noReconnect();
 
-            /**
+            /*
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/NatsAutoBench.java:65]: If this was around an api javadoc and maybe broke the doc, then absolutely, but again, Non client production code, only if already changing or fixing the class.

File: src/examples/java/io/nats/examples/autobench/ThrottledBenchmark.java [hunk 9]
```diff
@@ -73,10 +73,8 @@ public void execute(Options connectOptions) throws InterruptedException {
      * become slow consumers. The addition of this code was motivated by the asymetry in the java library between
      * publishers and subscribers, especially if UTF-8 subjects are enabled.
      * @param nc the connection
-     * @throws InterruptedException
      */
-	void adjustAndSleep(Connection nc) throws InterruptedException {
-
+	void adjustAndSleep(Connection nc) {
```
Comment (scottf) [inline src/examples/java/io/nats/examples/autobench/ThrottledBenchmark.java:77]: Non client production code, only if already changing or fixing the class. BUT, this is an example, if you wanted to make a separate PR for just this, then I would go so far as to properly fix the exception handling too.

File: src/examples/java/io/nats/examples/chaosTestApp/ConnectableConsumer.java [hunk 10]
```diff
@@ -37,7 +37,7 @@ public abstract class ConnectableConsumer implements ConnectionListener {
     protected String durableName;
     protected String label;
 
-    public ConnectableConsumer(CommandLine cmd, String initials, ConsumerKind consumerKind) throws IOException, InterruptedException, JetStreamApiException {
+    public ConnectableConsumer(CommandLine cmd, String initials, ConsumerKind consumerKind) throws IOException, InterruptedException {
```
Comment (scottf) [inline src/examples/java/io/nats/examples/chaosTestApp/ConnectableConsumer.java:40]: Non client production code, only if already changing or fixing the class. BUT... This could be cleaned up if you were already working on this and in it's own PR.

File: src/examples/java/io/nats/examples/chaosTestApp/Output.java [hunk 11]
```diff
@@ -63,7 +63,7 @@ public enum Screen {Left, Main, Console}
         String fontName = Font.MONOSPACED;
         GraphicsEnvironment localEnv;
         localEnv= GraphicsEnvironment.getLocalGraphicsEnvironment();
-        String allfonts[] = localEnv.getAvailableFontFamilyNames();
+        String[] allfonts = localEnv.getAvailableFontFamilyNames();
```
Comment (scottf) [inline src/examples/java/io/nats/examples/chaosTestApp/Output.java:66]: Non client production code, only if already changing or fixing the class.

**Review by scottf — CHANGES_REQUESTED** [review]
We appreciate the effort here, but it's not prudent to accept a PR with so many file changes.

I stopped making comments because they were mostly repeating themselves. Basically they fell into some categories.

* If you are already working with some code, then fine, it's okay to change this, but not to change it just to change on this scale.
* Non client production code, only if already changing or fixing the class.
* Not really that important, only if already changing or fixing the class.
* Unless you are aware of the intention of the author and comment, do not add paragraphing just to add it.

If you want to continue with this, fine, but you need smaller PRs. I would group them somehow. Package, functional area, just one class, etc.

**Comment by scottf** [comment]
Closing as per review comments.

**Comment by cbasener** [comment]
ok, I got it.
I will start small 😃
